macros/APIListAlpha.ejs

```diff
@@ -1,45 +1,65 @@
 <%
-// Creates a list of API Interfaces sorted alphabeticlly
+// Creates a list of API Interfaces sorted alphabetically
 
-var APIHref = $0 || '/en-US/docs/Web/API';
+const APIHref = $0 || '/en-US/docs/Web/API';
+
+/**
+ * Checks if the page is an interface page.
+ *
+ * @param {string} pageSlug
+ * @param {string[]} [tags]
+ */
+function isInterface(pageSlug, tags = null) {
+    const spaceIndex = pageSlug.indexOf('_');
 
-function isInterface(pageSlug) {
-    var spaceIndex = pageSlug.indexOf('_');
-    
     // If there are no spaces (represented by underscores), then we can
     // just return true, since this is by our rules an interface in that
-    // situation. Unless the slug is simply "Reference" (why the exception?).
-    
+    // situation.
+
+    // Unless the slug is simply "Reference", which is because that page
+    // used to host this macro, until it was moved to "Web/API".
+
     if ((spaceIndex == -1) && (pageSlug != "Reference")) {
         return true;
     }
-    
+
     // If the character after the space is '(', then this is assumed to be an
     // interface with a qualifier after its name (such as "(Firefox OS)").
     // These are allowed to appear in the index.
-    
+
     if ((spaceIndex < pageSlug.length-1) && (pageSlug[spaceIndex+1] == '(')) {
         return true;
     }
-    
-    // If we get to here, then it's not an interface.
-    
-    return false;
-}
 
-// Prepares the title of the link. This includes wrapping the appropriate
-// portion of the title with <code>. Returns null if the page should not
-// be included in the index.
+    // If we get to here, then it's probably not an interface.
 
-function buildTitle(base) {
+    return containsTag(tags, 'Interface');
+}
+
+/**
+ * Prepares the title of the link. This includes wrapping the appropriate
+ * portion of the title with `<code>`.
+ *
+ * @param {string} base The page title
+ * @param {string} slug The page slug without the parent
+ *
+ * @return {[string, string]}
+ *         The first entry is the new appropriate portion wrapped with `<code>`.
+ *         The second entry is the raw title text.
+ */
+function buildTitle(base, slug) {
+    base = base.trim();
     var spaceIndex = base.indexOf(' ');
-    
+
     // If there are no spaces in the page's title, the entire base
     // string is the title, so we just wrap the whole thing in <code> and
     // return it.
-    
-    if ((spaceIndex == -1) && (base != "Reference")) {
-        return "<code>" + base + "</code>";
+
+    if (spaceIndex == -1) {
```

Comment:
We no longer have to check if the page title is `Reference` here, as we already determine that in the `isInterface` function using the slug, which is more reliable.